tutorials/2023-11-06-offchain-sdk-ts-generation-with-cpm/index.md

```diff
@@ -0,0 +1,110 @@
+defaults:
+  contract-source-network: mainnet
+  contract-generate-sdk: true
+  contract-download: false
+  off-chain:
+    languages:
+    - ts
+    destinations:
+      ts: src/contracts
+
+contracts:
+  - label: NEO token
+    script-hash: '0xef4073a0f2b305a38ec4050e4d3d28bc40ea63f5'
+
+networks:
+  - label: mainnet
+    hosts:
+      - 'https://mainnet1.neo.coz.io:443'
+      - 'http://seed1.neo.org:10332'
+```
+
+> For more information on how to configure `cpm.yaml`, check out [CPM's documentation](https://github.com/CityOfZion/cpm/blob/master/docs/config.md)
+
+Then just run `cpm run` on the terminal and the SDK should be generated at the `src/contracts/neotoken` folder.' Anytime the contract adds or removes method signatures, you'll have to run `cpm run` again to update your SDK.
+
+## 3. Initializing and running the SDK
+Now that the SDK is already generated, you'll need to install the dependencies: [`@types/node`](https://www.npmjs.com/package/@types/node), and  [`@cityofzion/neon-dappkit-types`](https://www.npmjs.com/package/@cityofzion/neon-dappkit-types). Then you'll have to import and initialize the SDK to be able to call it.
+To initialize the SDK, you'll first need to instantiate a class that implements `Neo3-Invoker`.
+To enhance security and user experience, it's advisable to use [`WalletConnect`](https://github.com/CityOfZion/wallet-connect-sdk) for scenarios requiring user authentication, such as signing transactions or accessing user account details. However, for situations where such authentication is unnecessary, [`NeonInvoker`](https://github.com/CityOfZion/neon-dappkit/blob/main/packages/neon-dappkit/src/NeonInvoker.ts) from NeonDappkit offers a more streamlined solution, as it eliminates the need for users to approve requests through their wallets. In our example, we employ both methods: NeonInvoker is utilized when the user is not logged into their wallet.
+```ts
+import { NeoToken } from './contracts/neotoken'
+
+import { Neo3Invoker } from "@cityofzion/neon-dappkit-types"
+import { NeonInvoker } from "@cityofzion/neon-dappkit"
+import WcSdk from '@cityofzion/wallet-connect-sdk-core'
+import SignClient from '@walletconnect/sign-client'
+
+let wcSdk: WcSdk
+let neonInvoker: NeonInvoker
+
+async function initInvokers(): Promise<NeoToken> {
+  let signClient
+
+  signClient = await SignClient.init({
+    projectId: 'a9ff54e3d56a52230ed8767db4d4a810', // the ID of your project on Wallet Connect website
+    relayUrl: 'wss://relay.walletconnect.com', // we are using walletconnect's official relay server
+    metadata: {
+    name: 'MyApplicationName', // your application name to be displayed on the wallet
+    description: 'My Application description', // description to be shown on the wallet
+    url: 'https://myapplicationdescription.app/', // url to be linked on the wallet
+    icons: ['https://myapplicationdescription.app/myappicon.png'], // icon to be shown on the wallet
+    },
+  })
+  wcSdk = new WcSdk(signClient)
```

Comment:
After fixing a [compatibility problem with signClient](https://app.clickup.com/t/86a1dpp9f) we changed the sdk initalization to pass the options directly (the older way is still possible, but depending on the signClient version, it will show type errors). [Reference](https://github.com/CityOfZion/wallet-connect-sdk/blob/main/packages/wallet-connect-sdk-core/README.md#initialize-the-client) 
```suggestion
let wcSdk: WcSdk
let neonInvoker: NeonInvoker

async function initInvokers(): Promise<NeoToken> {
  wcSdk = await WcSdk.Init({
    projectId: 'a9ff54e3d56a52230ed8767db4d4a810', // the ID of your project on Wallet Connect website
    relayUrl: 'wss://relay.walletconnect.com', // we are using walletconnect's official relay server
    metadata: {
      name: 'MyApplicationName', // your application name to be displayed on the wallet
      description: 'My Application description', // description to be shown on the wallet
      url: 'https://myapplicationdescription.app/', // url to be linked on the wallet
      icons: ['https://myapplicationdescription.app/myappicon.png'], // icon to be shown on the wallet
    },
  })
```